### PR TITLE
[PLAY-79] Provide text highlight example in Typeahead kit

### PR DESCRIPTION
**Pull request by thestephenmarshall**
1. Allows `components` prop to be extended.
1. Provides example of text highlighting using `Typeahead` kit.

____

#### Screens

Screenshot 2022-12-21 at 1 31 43 PM

Screenshot 2022-12-21 at 1 32 02 PM

#### Breaking Changes

No

#### Runway Ticket URL

N/A

#### Checklist:

- [ ] **LABELS** Add a label: `enhancement`, `bug`, `improvement`, `new kit`, `deprecated`, or `breaking`. See [Changelog & Labels](https://github.com/powerhome/playbook/wiki/Changelog-&-Labels) for details.
- [ ] **DEPLOY** Please add the `Milano` label when you are ready for a review.
- [ ] **SCREENSHOT** Please add a screen shot or two.
- [ ] **SPECS** Please cover your changes with specs.
- [ ] **READ DOCS** Please make sure you have read and understand the [Playbook Release Process](https://github.com/powerhome/playbook/wiki/Playbook-Releases)

### CODE REVIEW

File: playbook/app/pb_kits/playbook/pb_typeahead/_typeahead.jsx [hunk 1]
```diff
@@ -44,7 +45,16 @@ type TypeaheadProps = {
...
+const Typeahead = ({
+  async,
+  components = {},
+  createable,
+  getOptionLabel,
+  getOptionValue,
+  id,
+  loadOptions = noop,
+  ...props
+}: TypeaheadProps) => {
```
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/pb_typeahead/_typeahead.jsx:57]: This is just me using the alphabet.

File: playbook/app/pb_kits/playbook/pb_typeahead/_typeahead.jsx [hunk 2]
```diff
@@ -57,6 +67,7 @@ const Typeahead = ({ loadOptions = noop, getOptionLabel, id, getOptionValue, cre
       Option,
       Placeholder,
       ValueContainer,
+      ...components
```
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/pb_typeahead/_typeahead.jsx:70]: This is the actual magic that allows us to overload our default components for the one created in the example.

File: playbook/app/pb_kits/playbook/pb_typeahead/docs/_typeahead_with_highlight.jsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        <FlexItem>
+          <Title size={4}><span dangerouslySetInnerHTML={{ __html: highlighted(name) }} /></Title>
+          <Body color="light">
+            <span dangerouslySetInnerHTML={{ __html: highlighted(title) }} />{" • "}
+            {territory}
+          </Body>
+        </FlexItem>
+      </Flex>
+    )
+  }
```
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/pb_typeahead/docs/_typeahead_with_highlight.jsx:69]: This sure would be a lot easier if we could use our `User` kit, however it does not allow us to send JSX for the string props necessary to do so. This is me sorta building it from scratch without applying the wrapper class.

File: playbook/app/pb_kits/playbook/pb_typeahead/docs/_typeahead_with_highlight.jsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    SingleValue: ({ data }: any) => (
+      <span>{data.name}</span>
+    )
+  }
+
+  return (
+    <React.Fragment>
+      <Typeahead
+          components={customComponents}
+          formatOptionLabel={formatOptionLabel}
```
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/pb_typeahead/docs/_typeahead_with_highlight.jsx:84]: `components` allows us to overload the default components we have created in the kit which is essentially replacing `react-select` components as mentioned here: https://react-select.com/components#replaceable-components

`formatOptionLabel` is where we can add the `<mark>` tags to highlight the matching text.

File: playbook/app/pb_kits/playbook/pb_typeahead/docs/_typeahead_with_highlight.jsx [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+          getOptionLabel={(option) => option.name}
+          getOptionValue={(option) => option.title}
```
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/pb_typeahead/docs/_typeahead_with_highlight.jsx:86]: These are our search fields. Above, inside `formatOptionLabel` we receive whatever value we have formatted here.

**Comment by thestephenmarshall** [comment]
Provides some insight to #1389

**Comment by thestephenmarshall** [comment]
Currently, this is in draft mode due to this error where the last item is always selected per the CSS yet it isn't truly selected.

Screenshot 2022-12-21 at 1 32 10 PM

**Review by nidaqg — APPROVED** [review]

**Comment by thestephenmarshall** [comment]
Fixes #2231

**Review by jasperfurniss — APPROVED** [review]

**Pull request merged**